### ISSUE TITLE
Vitest

### ISSUE THREAD

**Issue by tisto** [plone/volto#6326]
## Abstract

Replace Jest with Vitest as the Volto unit test runner.

## Motivation

Jest is showing its age compared to Vitest. Vitest is the unit test runner that uses Vite under the hood. It uses a zero configuration model, thanks to be Vite-based. It's faster as it is Vite who transpiles the code.

## Assumptions

None.

## Proposal & Implementation

Vitest shares the same syntax as they use Mocha/Chai we will have to adapt the test setup to meet Vitest requirements and move the mocks to the Vitest style, if needed.

## Deliverables

Volto will be using Vitest instead of Jest for running tests.
Documentation on how to migrate projects and add-ons to use Vitest.
Documentation on Vitest.

## Risks

None.

## Participants

Open for volunteers. This could be easily a GSOC for next year.

**Comment by Abhishek-17h** [comment]
I am working on this issue. It requires a systematic approach for migration, and I will provide regular updates on my progress.

**Comment by Abhishek-17h** [comment]
I am halfway through the migration process and will raise a draft PR for corrections and suggestions.

**Comment by Shyam-Raghuwanshi** [comment]
@stevepiercy we need to do more discussion on this or should i proceed?

**Comment by stevepiercy** [comment]
@Shyam-Raghuwanshi I don't know why you `@` me. I'm not participating in this PLIP.

Please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do) and [Work with GitHub issues](https://6.docs.plone.org/contributing/first-time.html#work-with-github-issues).

**Comment by Abhishek-17h** [comment]
@Shyam-Raghuwanshi I am already working on it and u can take other issues that are open or not claimed to avoid any duplication and waste of effort.

**Comment by Shyam-Raghuwanshi** [comment]
@Abhishek-17h you should make a draft pr

**Comment by stevepiercy** [comment]
> [@Abhishek-17h](https://github.com/Abhishek-17h) you should make a draft pr

They should have followed the advice in https://github.com/plone/volto/issues/6326#issuecomment-2597723741. But since they did not follow that document and started work on this PLIP, publishing a draft PR is also a good suggestion. At least they would get feedback on whether they are on the right path, and possibly avoid wasting their time and effort.

I don't know how to help first-timers avoid spending time that might become a waste, other than what we have already set up. If you have other suggestions, please come to the [Documentation channel in Discord](https://discord.com/channels/786421998426521600/862642144401031188), and let's chat.

**Comment by Abhishek-17h** [comment]
> I am halfway through the migration process and will raise a draft PR for corrections and suggestions.

@Shyam-Raghuwanshi  yes, I will raise a draft PR for this soon and I already mentioned it.

**Comment by alihamza1221** [comment]
do  it draft mean draft what you want from, I want to be truthy here you
are actually blocking other contributors to work. Looks like you are afraid
of something.
Believe this is not how this works, if i had some more time would have
worked this just because of your behaviour and had ruined your dreams.

**Comment by Abhishek-17h** [comment]
@alihamza1221 hey , I am working on this issue actually over a month and volto is very large repo and migration is not straight forward and it takes time. I am not afraid of anything and U should understand if someone is working on some issue u should take another issue ,so it don't waste efforts of both contributors.In plone they don't assign issue, u have to claim it and work on that and that's what I am doing. Hope u have gone through plone's documentation and documents mentioned here

> [@Shyam-Raghuwanshi](https://github.com/Shyam-Raghuwanshi) I don't know why you `@` me. I'm not participating in this PLIP.
> 
> Please read and follow [First-time contributors](https://6.docs.plone.org/contributing/first-time.html), especially [Things not to do](https://6.docs.plone.org/contributing/first-time.html#things-not-to-do) and [Work with GitHub issues](https://6.docs.plone.org/contributing/first-time.html#work-with-github-issues).
